src/Stratis.Bitcoin.Features.BlockStore/AddressIndexing/AddressIndexerOutpointCache.cs

```diff
@@ -0,0 +1,169 @@
+        {
+            public LRUItem(string outPoint, OutPointData outPointData)
+            {
+                Guard.NotEmpty(outPoint, nameof(outPoint));
+                Guard.NotNull(outPointData, nameof(outPointData));
+
+                this.Key = outPoint;
+                this.Value = outPointData;
+            }
+
+            public readonly string Key;
+
+            public readonly OutPointData Value;
+        }
+
+        public const int AddressIndexOutputCacheMaxItemsDefault = 100000;
+
+        private const int BatchSize = 100;
+
+        /// <summary>The maximum number of items that can be kept in the cache until
+        /// entries will start getting evicted.</summary>
+        public readonly int MaxItems;
+
+        public int Count => this.outPointLinkedList.Count;
+
+        /// <summary>A mapping between the string representation of an outpoint and its
+        /// corresponding scriptPubKey and money value.
+        /// All access to the cache must be protected with <see cref="lockObj"/>.</summary>
+        private readonly Dictionary<string, LinkedListNode<LRUItem>> cachedOutPoints;
+
+        /// <summary>A linked list used to efficiently determine the oldest entry
+        /// in the cache. Items get added to the cache in time order and get moved to
+        /// the back of the eviction list each time they are accessed.</summary>
+        private readonly LinkedList<LRUItem> outPointLinkedList;
+
+        private object lockObj;
+
+        private readonly LiteDatabase db;
+
+        private readonly LiteCollection<OutPointData> addressIndexerOutPointData;
+
+        public AddressIndexerOutpointCache(LiteDatabase db, string addressIndexerOutputCollectionName, int maxItems = 0)
+        {
+            this.lockObj = new object();
+            this.db = db;
+            this.addressIndexerOutPointData = this.db.GetCollection<OutPointData>(addressIndexerOutputCollectionName);
+
+            this.MaxItems = maxItems == 0 ? AddressIndexOutputCacheMaxItemsDefault : maxItems;
+
+            this.cachedOutPoints = new Dictionary<string, LinkedListNode<LRUItem>>();
+            this.outPointLinkedList = new LinkedList<LRUItem>();
+        }
+
+        public void AddToCache(OutPointData outPointData)
+        {
+            lock (this.lockObj)
+            {
+                // Don't bother adding the entry if it exists already.
+                if (this.cachedOutPoints.ContainsKey(outPointData.Outpoint))
+                    return;
+
+                if (this.cachedOutPoints.Count >= this.MaxItems)
+                    this.EvictOldest();
+
+                var item = new LRUItem(outPointData.Outpoint, outPointData);
+                var node = new LinkedListNode<LRUItem>(item);
+                this.outPointLinkedList.AddLast(node);
+                this.cachedOutPoints.Add(outPointData.Outpoint, node);
+            }
+        }
+
+        /// <summary>When an output is spent there is no point retaining it
```

Comment:
why using 2 lines? can be done on 1 line, we all have big monitors to fit 160 characters. Same for param